analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,515 @@
+}
+```
+
+## Combine arm, cytoband, and gene-level status data
+
+```{r}
+# The logic variable `running_in_ci` is needed here because the CI testing
+# files do not contain any of the genes in the `gene_status_count` data.frame
+# generated above (when `running_in_ci` == FALSE)
+if (!(running_in_ci)) {
+  combined_status_count_df <- consensus_seg_autosomes_df %>%
+    mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband)) %>%
+    inner_join(arm_status_count,
+               by = "chromosome_arm") %>%
+    left_join(cytoband_status_count,
+              by = "cytoband",
+              suffix = c(".arm", ".cytoband")) %>%
+    left_join(gene_status_count,
+              by = "gene_symbol",
+              suffix = c(".arm", ".gene")) %>%
+    mutate(
+      focal_call = paste0(
+        dominant_status.arm,
+        ", ",
+        dominant_status.cytoband,
+        ", ",
+        dominant_status
+      ),
+      # Here we want to define the most focal call based on the arm, cytoband,
+      # and gene status information -- if a loss/gain is defined at the arm
+      # level then the focal call will be "arm_loss" or "arm_gain" respectively,
+      # and so on.
+      focal_call = case_when(focal_call == "loss, NA, NA" ~ "arm_loss",
+                             focal_call == "neutral, uncallable, NA" ~ "uncallable",
+                             focal_call == "neutral, loss, NA" ~ "cytoband_loss",
+                             focal_call == "neutral, NA, NA" ~ "arm_neutral",
+                             focal_call == "neutral, unstable, neutral" ~ "gene_neutral",
+                             focal_call == "neutral, unstable, loss" ~ "gene_loss",
+                             TRUE ~ "Other")
+    )
+} else {
+  combined_status_count_df <- consensus_seg_autosomes_df %>%
+    mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband)) %>%
+    inner_join(arm_status_count, by = "chromosome_arm") %>%
+    mutate(focal_call = dominant_status)
+}
+
+# Display combined status counts table
+combined_status_count_df %>%
+  arrange(focal_call) %>%
+  group_by(biospecimen_id, focal_call)
+```
+
+## Transform data into long format
+
+### Tranform the chromosome arm status data
+
+```{r}
+if (!(running_in_ci)) {
+  final_arm_status_df <- combined_status_count_df %>%
+    # Filter to only non-neutral chromosome arms
+    filter(dominant_status.arm != "neutral") %>%
+    select(
+      Kids_First_Biospecimen_ID = biospecimen_id,
+      region = chromosome_arm,
+      status = dominant_status.arm
+    ) %>%
+    distinct() %>%
+    mutate(region_type = "chromosome_arm")
+} else {
+  final_arm_status_df <- combined_status_count_df %>%
+    # Filter to only non-neutral chromosome arms
+    filter(dominant_status != "neutral") %>%
+    select(
+      Kids_First_Biospecimen_ID = biospecimen_id,
+      region = chromosome_arm,
+      status = dominant_status
+    ) %>%
+    distinct() %>%
+    mutate(region_type = "chromosome_arm")
```

Comment:
You can get rid of these mutate steps, if at your `bind_rows` step at 479, you use `.id` argument.